lib/samples/edit_geometries_with_programmatic_reticle_tool/edit_geometries_with_programmatic_reticle_tool.dart

```diff
@@ -0,0 +1,566 @@
+    return Scaffold(
+      body: SafeArea(
+        left: false,
+        right: false,
+        child: Stack(
+          children: [
+            Column(
+              children: [
+                // Creates the top menu used for adjusting settings in the geometry editor.
+                buildTopMenu(),
+                Expanded(
+                  // Add a map view to the widget tree and set a controller.
+                  child: ArcGISMapView(
+                    controllerProvider: () => _mapViewController,
+                    onMapViewReady: onMapViewReady,
+                    // Configure interactions with the map view using the onTap callback.
+                    onTap: onTap,
+                  ),
+                ),
+                Padding(
+                  padding: const EdgeInsets.all(10),
+                  child: Row(
+                    mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                    children: [
+                      // Creates the button which controls the current action of the geometry editor.
+                      Expanded(child: buildGeometryEditorActionButton()),
+                    ],
+                  ),
+                ),
+              ],
+            ),
+          ],
+        ),
+      ),
+      // A bottom sheet that shows/hides settings relating to the geometry editor depending on the value of the showSettings flag.
+      bottomSheet: _showSettings ? buildBottomSheet() : null,
+    );
+  }
+
+  void onMapViewReady() {
+    // Create a map with the imagery basemap style and set to the map view controller.
+    final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISImageryStandard);
+    _mapViewController.arcGISMap = map;
+
+    // Configure some initial graphics and add to the graphics overlay.
+    _graphicsOverlay.graphics.addAll(initialGraphics());
+    // Add the graphics overlay to the map view.
+    _mapViewController.graphicsOverlays.add(_graphicsOverlay);
+
+    // Set an initial viewpoint over the graphics.
+    _mapViewController.setViewpoint(
+      Viewpoint.fromCenter(
+        ArcGISPoint(
+          x: -0.775395,
+          y: 51.523806,
+          spatialReference: SpatialReference.wgs84,
+        ),
+        scale: 20000,
+      ),
+    );
+
+    // Set the programmatic reticle tool as the active tool on the geometry editor.
+    _geometryEditor.tool = _programmaticReticleTool;
+    // Listen to changes in the geometry editor to manage the UI.
+    // When the picked up element changes, we update the state.
+    _geometryEditor.onPickedUpElementChanged.listen(
+      (pickedUpElement) =>
+          setState(() => _hasPickedUpElement = pickedUpElement != null),
+    );
+    // When the hovered element changes, we update the state.
+    _geometryEditor.onHoveredElementChanged.listen(
+      (hoveredElement) => setState(() => _hoveredElement = hoveredElement),
+    ); // Listen to changes in canUndo and canRedo in order to enable/disable the UI.
```

Comment:
I think this comment was meant to be on its own separate line

Reply:
Good catch